src/xcsync/Workers/ChangeWorker.cs

```diff
@@ -21,7 +21,7 @@ public override async Task ConsumeAsync (ChangeMessage message, CancellationToke
 		if (message.Direction == SyncDirection.ToXcode)
 			XcodeCommand<GenerateCommand>.RecreateDirectory (FileSystem, TargetDir);
 		Logger.Debug (Strings.Watch.Syncing);
-		await new SyncContext (FileSystem, TypeService, message.Direction, ProjectPath, TargetDir, Framework, Logger).SyncAsync (cancellationToken);
+		await new SyncContext (FileSystem, new TypeService (Logger!), message.Direction, ProjectPath, TargetDir, Framework, Logger).SyncAsync (cancellationToken);
```

Comment:
Whenever I see a `!` assertion, I ask if this is avoidable.
In this case, I see the `Logger` variable being used several time before the assertion. This tells me that if `Logger` is null, this code will crash with a NRE before getting to the `!`.
I did a quick check on the source to serilog and it looks like there's no protection on the method. For example, if you implemented `Debug` like this:
```
public static void MyDebug(this ILogger? logger, string message)
{
    if (logger is null) return; // or exception?
    logger.Debug(message);
}
```
But it's not - it's a straight method.
What I would consider is:
1. put `#nullable enable` on this file and refactor appropriately.
2. if it makes sense for the `ILogger` to be null, then handle it appropriately before the `Debug` calls, the analyzer should be able to see that and not issue a warning so you shouldn't need the `!`; if not then the `!` isn't needed.